### PR TITLE
Adding an option to allow full-path patterns in pytest.

### PR DESCRIPTION
**Pull request by thirtytwobits**
I'd like to undo my kluge here: https://github.com/UAVCAN/nunavut/blob/master/conftest.py#L335

If you accept this patch it shouldn't change any existing uses. This is an "opt-in" change.

### CODE REVIEW

**Comment by cjw296** [comment]
I can't really follow what the intention of this change is or why you need it. 
This, and the CircleCI, are a good indication that some tests that show your use case need to be added to this PR.

**Comment by thirtytwobits** [comment]
I can add tests if you agree with the approach. If you look at the link (https://github.com/UAVCAN/nunavut/blob/master/conftest.py#L335) you'll see we use full glob patterns that include paths to include/exclude files by folder instead of just by a simple filename. This change allows us to do that without implementing our own pytest hook.

## TLDR:

### Supported Today
```
*.py
```

### With This Change

```
**/foo/**/*.py
```

**Comment by cjw296** [comment]
I think https://github.com/cjw296/sybil/pull/23 may be what you're after.

I also wanted to make you aware that the high level of snark in [your code](https://github.com/UAVCAN/nunavut/blob/20b60ce4053db1d9799e1626841542c038c7c676/conftest.py#L322-L358) and your over-bolded TLDR above don't really encourage me to work on this on my free time.

It feels like you've misunderstood that Sybil's current support is for a directory of files, not a tree of directories and files.
However, the latter looks like an interesting use case, so I've worked up a PR to support it. I look forward to seeing what your thoughts are...

**Comment by thirtytwobits** [comment]
Hey Chris, sorry. The snark was directed at myself after struggling to get this to work over an hour one Friday evening and perhaps some of my frustration was misdirected. Nothing against you and your work I assure you! Just the opposite, I tell everybody about Sybil as an example of how one should be writing tests and documentation. I've used your work on two projects now, https://nanaimo.readthedocs.io/en/latest/README.html and https://nunavut.readthedocs.io/en/latest/README.html and will probably use it a lot more going forward. Also:

https://github.com/UAVCAN/nunavut/pull/121

#23 seems like it will add support for my usecase. Sorry. Thanks. Sorry.

**Comment by cjw296** [comment]
1.4.0 is up on pypi, let me know if it works for you!

**Comment by thirtytwobits** [comment]
> 1.4.0 is up on pypi, let me know if it works for you!

Works beautifully! Thanks so much.
